apps/server/src/infra/access-token/domain/access-token.service.spec.ts

```diff
@@ -0,0 +1,164 @@
+			});
+		});
+
+		describe('when storageClient.set rejects', () => {
+			const setup = () => {
+				const payload = { foo: 'bar' };
+				const tokenTtl = 1234;
+				const error = new Error('Storage error');
+
+				storageClient.set.mockRejectedValueOnce(error);
+
+				return { payload, tokenTtl, error };
+			};
+
+			it('should throw an error', async () => {
+				const { payload, tokenTtl, error } = setup();
+
+				await expect(service.createToken(payload, tokenTtl)).rejects.toThrow(error);
+				expect(storageClient.set).toHaveBeenCalledWith(expect.any(String), JSON.stringify(payload), 'EX', tokenTtl);
+			});
+		});
+	});
+
+	describe('resolveToken', () => {
+		describe('when storageClient.get resolves', () => {
+			describe('when token exists and is valid JSON', () => {
+				const setup = () => {
+					const token = randomUUID();
+					const tokenTtl = 1234;
+					const payload = { foo: 'bar' };
+					const value = JSON.stringify(payload);
+
+					storageClient.get.mockResolvedValueOnce(value);
+
+					return { token, tokenTtl, payload, value };
+				};
+
+				it('should return the parsed payload and renew the token timeout', async () => {
+					const { token, tokenTtl, payload } = setup();
+
+					const result = await service.resolveToken({ token, tokenTtl });
+
+					expect(storageClient.get).toHaveBeenCalledWith(token);
+					expect(storageClient.set).toHaveBeenCalledWith(expect.any(String), JSON.stringify(payload), 'EX', tokenTtl);
+					expect(result).toEqual(payload);
+				});
+			});
+
+			describe('when token does not exist', () => {
+				const setup = () => {
+					const token = 'token-uuid';
+					const tokenTtl = 1234;
+					const error = new ForbiddenException(`Token ${token} not found`);
+
+					storageClient.get.mockResolvedValueOnce(null);
+
+					return { token, tokenTtl, error };
+				};
+
+				it('should throw ForbiddenException', async () => {
+					const { token, tokenTtl, error } = setup();
+
+					await expect(service.resolveToken({ token, tokenTtl })).rejects.toThrow(error);
+					expect(storageClient.get).toHaveBeenCalledWith(token);
+				});
+			});
+
+			describe('when token value is invalid JSON', () => {
+				const setup = () => {
+					const token = 'token-uuid';
+					const tokenTtl = 1234;
+					const value = 'not-json';
+					const jsonError = new Error(`Unexpected token 'o', \"not-json\" is not valid JSON`);
+					const error = new InternalServerErrorException(`Invalid payload for token ${token}`, { cause: jsonError });
+
+					storageClient.get.mockResolvedValueOnce(value);
+
+					return { token, tokenTtl, value, error };
+				};
+
+				it('should throw InternalServerErrorException', async () => {
+					const { token, tokenTtl, error } = setup();
+
+					await expect(service.resolveToken({ token, tokenTtl })).rejects.toThrowError(error);
+					expect(storageClient.get).toHaveBeenCalledWith(token);
+				});
+			});
+		});
+
+		describe('when storageClient.get rejects', () => {
```

Comment:
Das when ist hier noch etwas verloren, muss eingerückt, oder zugeordnet werden.

Reply:
Ahh ok, da drüber sind 2 when in einander geschachtelt. 
Wie wäre es mit:
'given storageClient.get resolves'
'given storageClient.get rejects'